reference/src/taimpl/test/ta_test_helpers.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020-2021 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef TA_TEST_HELPERS_H
+#define TA_TEST_HELPERS_H
+
+#include "ta_sa.h"
+#include "test_helpers.h"
+#include <memory>
+#include <vector>
+
+#define MAX_CLIENT_SLOTS 256
+
+namespace ta_test_helpers {
+    using namespace test_helpers;
+
+    /**
+     * Returns the UUID of the "test" TA.
+     *
+     * @return the UUID of the "test" TA.
+     */
+    const sa_uuid* ta_uuid();
+
+    /**
+     * Returns the test TA client.
+     *
+     * @return the test TA client.
+     */
+    ta_client client();
+
+    /**
+     * Creates a shared pointer holding the sa_key value. Key will be released using
+     * sa_key_release when the shared pointer gets destroyed.
+     *
+     * @return created shared pointer.
+     */
+    std::shared_ptr<sa_key> create_uninitialized_sa_key();
+
+    /**
+     * Creates a shared pointer holding the sa_crypto_cipher_context value. Context will be
+     * released using sa_crypto_cipher_release when the shared pointer gets destroyed.
+     *
+     * @return created shared pointer.
+     */
+    std::shared_ptr<sa_crypto_cipher_context> create_uninitialized_sa_crypto_cipher_context();
+
+    /**
+     * Allocates an sa_buffer with the given type and size.
+     *
+     * @param buffer_type the type of the buffer.
+     * @param size the size of the buffer.
+     * @return the buffer.
+     */
+    std::shared_ptr<sa_buffer> buffer_alloc(
+            sa_buffer_type buffer_type,
+            size_t size);
+
+    /**
+     * Allocates an sa_buffer with the given type and initializes it with the given value.
+     *
+     * @param buffer_type the type of the buffer.
```

Comment:
Should include param direction as done in other areas of the code.

Reply:
Done